scripts/coverage.sh

```diff
@@ -37,7 +38,9 @@ rm -rf target/cov/$reportName
 
 source ci/rust-version.sh nightly
 # shellcheck disable=SC2086 #
-_ cargo +$rust_nightly test --target-dir target/cov --lib $crate
+RUST_LOG=solana=trace _ cargo +$rust_nightly test --target-dir target/cov --lib --no-run $crate "$@"
+# shellcheck disable=SC2086 #
+RUST_LOG=solana=trace _ cargo +$rust_nightly test --target-dir target/cov --lib $crate "$@" 2> /dev/null
```

Comment:
Separated compile and test for easier debugging because the messages from `cargo build` would also discarded altogether otherwise.

Reply:
Why hide stderr here? `2> /dev/null`?

Reply:
Otherwise, tons of tons of logs will be printed on the terminal, after all we're running all test with **trace** level. Maybe, better off to note that as a comment?

Or, should we stash the logs to `2> $(mktemp)`? I think this is overdue because we're in the coverage run. It's unlikely we need to debug using this build.

Reply:
Ah, I see.  Yes there are times when the build only fails in a coverage run in CI (we had an occurrence  of this today in fact!) so being able to view the log could be helpful.

We don't need anything fancy like `mktemp` here though.  Instead just redirect to a fixed file, `coverage-stderr.log`, then use https://github.com/solana-labs/solana/blob/master/ci/upload-ci-artifact.sh to include the log file as a buildkite artifact for easy access by anybody who wants it

Reply:
Thanks for explanation done https://github.com/solana-labs/solana/pull/6345/commits/98005bf80f1eb46f69388beefc0577565a909a40..

By the way,  I'm guessing for the log file to be ranges of X00MB.. Let's see if its filesize is too big, or not... If that's concern, I'll just compress it via `bz2`.

Reply:
Yay, [build passed](https://buildkite.com/solana-labs/solana/builds/13174#c20406ba-f17d-4843-a01d-d3f64d1781a8).

Heh, I was pessimistic a bit, it seems.

@mvines Hi, are ~100MB artifacts a bit concern? Should compress?


```
artifact: target/cov/coverage-stderr.log
-rw-r--r-- 1 buildkite-agent buildkite-agent 94182845 Oct 16 19:03 target/cov/coverage-stderr.log
+ buildkite-agent artifact upload target/cov/coverage-stderr.log
2019-10-16 19:05:27 INFO   Found 1 files that match "target/cov/coverage-stderr.log"
2019-10-16 19:05:27 INFO   Creating (0-1)/1 artifacts
2019-10-16 19:05:28 INFO   Uploading artifact 8e17d76a-93ea-4883-99d9-3606292791b4 target/cov/coverage-stderr.log (94182845 bytes)
2019-10-16 19:05:31 INFO  Successfully annotated build
```

Reply:
Oh wow.  Sure please `gzip` it.  I don't expect this to be used very often so the extra decompression step for the user should be no big deal

Reply:
it really shrank!

```
-rw-r--r-- 1 buildkite-agent buildkite-agent 10056926 Oct 16 23:51 target/cov/coverage-stderr.log.gz
+ buildkite-agent artifact upload target/cov/coverage-stderr.log.gz
2019-10-16 23:52:52 INFO   Found 1 files that match "target/cov/coverage-stderr.log.gz"
2019-10-16 23:52:52 INFO   Creating (0-1)/1 artifacts
2019-10-16 23:52:52 INFO   Uploading artifact 98e02c81-9b3f-456d-95ee-4826660c7cb3 target/cov/coverage-stderr.log.gz (10056926 bytes)
2019-10-16 23:52:54 INFO  Successfully annotated build
```